qulice-pmd/src/test/java/com/qulice/pmd/PMDEmptyTest.java

```diff
@@ -0,0 +1,218 @@
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.qulice.pmd;
+
+import com.qulice.spi.Environment;
+import com.qulice.spi.ValidationException;
+import java.io.StringWriter;
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+import org.apache.log4j.SimpleLayout;
+import org.apache.log4j.WriterAppender;
+import org.hamcrest.Matcher;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link PMDValidator} class.
+ * @author Prahlad Yeri (prahladyeri@yahoo.com)
+ * @version $Id$
+ */
+@SuppressWarnings("PMD.TooManyMethods")
+public final class PMDEmptyTest {
+    /**
+     * Makes sure that empty static initializers fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyStaticInitializerTest() throws Exception {
+        this.validatePMD(
+            "EmptyStaticInitializer.java",
+            Matchers.containsString(
+                "Empty static initializer was found"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty statement blocks fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyStatementBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptyStatementBlock.java",
+            Matchers.containsString(
+                "Avoid empty block statements"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty initializers fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyInitializerTest() throws Exception {
+        this.validatePMD(
+            "EmptyInitializer.java",
+            Matchers.containsString(
+                "Empty initializer was found"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty statement not in a loop fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptyStatementNotInLoopTest() throws Exception {
+        this.validatePMD(
+            "EmptyStatementNotInLoop.java",
+            Matchers.containsString(
+                "An empty statement (semicolon) not part of a loop"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty synchronized statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptySynchronizedBlockTest() throws Exception {
+        this.validatePMD(
+            "EmptySynchronizedBlock.java",
+            Matchers.containsString(
+                "Avoid empty synchronized blocks"
+            )
+        );
+    }
+
+    /**
+     * Makes sure that empty switch statements fail with an error.
+     * @throws Exception when something goes wrong
+     */
+    @Test
+    public void emptySwitchStmtTest() throws Exception {
```

Comment:
@prahladyeri Let's call this method `failsForEmptySwitchStatement`, see [this page](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names) for more details.